### PR TITLE
Fix GitHub and Jira Integration 

### PR DESCRIPTION
**Pull request by RaiVaibhav**

### CODE REVIEW

**Comment by RaiVaibhav** [comment]
Hey, @cosmosgenius there was some error occurred in integrating the error flow with ember concurrency in the frontend, I have discussed with @Praseetha-KR, can't confirm whether that error is because of error handling flow compatibility of ember Adapter with Django REST API, can you please take a look.

Exact Error
```
TypeError: _private.AdapterError is not a constructor.
```
which was thrown during `try`/`catch` of ember-concurrency when I tried to do something like

```
try{
  this.get("store").findRecord("modeName", <id>) // when their is no record i.e., 404, response details:["not found"]
}catch(err){
  // err.payload.detail (no payload, instead it was a TypeError)
}
// `device-preferences` also thrown the same error. Status Code - 500

**Comment by cosmosgenius** [comment]
@RaiVaibhav can you rebase this?

File: app/components/github-project.js [hunk 1]
```diff
@@ -10,74 +11,139 @@ const GithubProjectComponent = Component.extend({
...
+      )
+    }
+    try{
+      let currentGithubRepo = yield githubRepo.save()
+      this.set("isChangingRepo", false);
+      this.set("currentGithubRepo",currentGithubRepo);
+      this.get("notify").success(tRepoIntegrated);
+      yield this.set("showEditGithubModal", false )
+    }catch(error){
+      yield githubRepo.rollbackAttributes();
```
[the remark this answers is not in the document]
Comment (RaiVaibhav) [reply app/components/github-project.js]: Do I create a new property to store the `githubRepo` and then use it as I need the `rollbackAttributes`, (try catch here will be needed if I don't store the githubRepo in some property.

File: app/components/github-account.js [hunk 2]
```diff
@@ -3,27 +3,45 @@ import { inject as service } from '@ember/service';
...
+      this.get("notify").error(this.get('tGithubErrorIntegration'));
+    }
+
+  }),
 
   confirmCallback() {
     const tGithubWillBeRevoked = this.get("tGithubWillBeRevoked");
     this.set("isRevokingGithub", true);
-    this.get("ajax").post(ENV.endpoints.revokeGitHub)
+    this.get("ajax").delete(`/api/organizations/${this.get('organization.selected.id')}/github`)
```
[the remark this answers is not in the document]
Comment (RaiVaibhav) [reply app/components/github-account.js]: If I made the confirmCallback task then `confirmCallback` need to be changed everywhere in the whole app to the task as it is getting used in `confirm-box` component. This change is related to the PR ? or should I make a different PR for this change?

**Pull request merged**